packages/flame/lib/src/collisions/quadtree/quadtree_collision_detection.dart

```diff
@@ -0,0 +1,105 @@
+import 'package:flame/collisions.dart';
+import 'package:flame/components.dart';
+import 'package:flutter/widgets.dart';
+
+/// Collision detection modification to support a Quad Tree broadphase.
+///
+/// Do not use standard [items] list for components. Instead adds all components
+/// into [QuadTreeBroadphase] class.
+class QuadTreeCollisionDetection extends StandardCollisionDetection {
+  QuadTreeCollisionDetection({
+    required Rect mapDimensions,
+    required ExternalBroadphaseCheck broadphaseCheck,
+    required ExternalMinDistanceCheck minimumDistanceCheck,
+    int maxObjects = 25,
+    int maxLevels = 10,
+  }) : super(
+          broadphase: QuadTreeBroadphase<ShapeHitbox>(
+            mainBoxSize: mapDimensions,
+            maxObjects: maxObjects,
+            maxLevels: maxLevels,
+            broadphaseCheck: broadphaseCheck,
+            minimumDistanceCheck: minimumDistanceCheck,
+          ),
+        );
+
+  QuadTreeBroadphase get quadBroadphase => broadphase as QuadTreeBroadphase;
+
+  final _listenerTransform = <ShapeHitbox, VoidCallback>{};
+  final _listenerCollisionType = <ShapeHitbox, VoidCallback>{};
+  final _scheduledUpdate = <ShapeHitbox>{};
+
+  @override
+  void add(ShapeHitbox item) {
+    super.add(item);
+    // ignore: prefer_function_declarations_over_variables
+    final listenerTransform = () {
+      if (item.isMounted) {
```

Comment:
As long as the item is getting added here, it should always be mounted right, so no need for this check?

Reply:
This is a listener. It might be called at any time, its probably a wrong function declaration confuses you.